internal/context.go

```diff
@@ -419,3 +426,15 @@ func (c *valueCtx) Value(key interface{}) interface{} {
 	}
 	return c.Context.Value(key)
 }
+
+func spanFromContext(ctx Context) opentracing.Span {
+	val := ctx.Value(activeSpanKey)
+	if sp, ok := val.(opentracing.Span); ok {
+		return sp
+	}
+	return nil
```

Comment:
Shouldn't we return NOOP Span instead?

Reply:
This is a copy of the open tracing implementation. Returns a nil in absence of an opentracing Span.

Reply:
Does it mean that all Span using code always checks for this nil before using a Span?

Reply:
Yes, but generally code (even RPC) doesnt need to deal with spans directly. They do something like.. 
```
span = opentracing.SpanFromContext(ctx)
newSpan = tracer.CreateSpan(ChildOf(span))
defer newSpan.Finish()
```

So the tracer is expected to handle that span can actually be nil..